.github/workflows/deploy.yml

```diff
@@ -173,6 +173,10 @@ jobs:
           path: '**/dist'
           key: dist-${{ env.rid }}
 
+      - name: Add NPM token to .yarnrc.yml file
+        run: |
+          echo "npmAuthToken: ${{ env.token }}" >> ~/.yarnrc.yml
```

Comment:
All the tooling changes that you see here are not part of this PR. While rebasing it seems something went wrong.
These changes are being done to fix the pipeline issues and we have not discussed with Tim regarding this as these fixes are being done because publish-npm was constantly failing.